datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -571,66 +580,73 @@ enum VisitRecord {
     /// `usize` is the monotone increasing series number assigned in pre_visit().
     /// Starts from 0. Is used to index the identifier array `id_array` in post_visit().
     EnterMark(usize),
+    /// the node's children were skipped => jump to f_up on same node
+    JumpMark(usize),
     /// Accumulated identifier of sub expression.
     ExprItem(Identifier),
 }
 
 impl ExprIdentifierVisitor<'_> {
-    fn desc_expr(expr: &Expr) -> String {
+    fn node_identifier(expr: &Expr) -> String {
```

Comment:
I renamed this, to make it more explicit that this is the identifier used in the `IdArray` and `ExprSet`.